### ISSUE TITLE
Tuple indexing causes printing error

### ISSUE THREAD

**Issue by weaversa** [GaloisInc/cryptol#490]
```
Main> [ (True, i) | i <- [0..1] ]
Assuming a = 1
[(True, 0), (True, 1)]
Main> [ (True, i) | i <- [0..1] ].0
[warning] at <interactive>:1:1--1:30:
  Defaulting type parameter 'bits'
             of finite enumeration
             at <interactive>:1:20--1:26
  to 1
[True, True]
```

`[True, True]` should be `3`.

**Comment by brianhuffman** [comment]
The interpreter uses a tag on vector values that says whether or not the elements have type `Bit`. I bet that the tag is being set incorrectly in this case.

**Comment by brianhuffman** [comment]
The offending code is in `src/Cryptol/Eval.hs`, function `evalSel`. The local functions `tupleSel` and `recordSel` both contain cases where the `VWord`/`VSeq` invariant can be violated. To prevent this, we need to test whether the resulting type of the tuple or record projection is `Bit`. Unfortunately, the interpreter doesn't have that type information easily available at that point, so we'd have to plumb it in from somewhere else, making this more than a 10-minute fix.

**Comment by weaversa** [comment]
Does this fall into the same category?
```
Main> zero : [inf]
[False, False, False, False, False, ...]
```
Maybe infinite sequences aways print at the level of the base type?

**Comment by brianhuffman** [comment]
You're right, infinite sequences are special and always print the elements individually.

**Comment by yav** [comment]
This should be fixed in 7bf0fa82221117f35d7eccae
